sklearn/ensemble/_forest.py

```diff
@@ -915,14 +971,27 @@ def _set_oob_score_and_attributes(self, X, y):
             The data matrix.
         y : ndarray of shape (n_samples, n_outputs)
             The target matrix.
+        sample_weight : ndarray of shape (n_samples,)
+            Sample weights.
         """
-        self.oob_prediction_ = super()._compute_oob_predictions(X, y).squeeze(
-            axis=1
+        self.oob_prediction_, oob_importances_ = (
+            super()
+            ._compute_oob_predictions_and_importances(X, y, sample_weight)
         )
+        self.oob_prediction_ = self.oob_prediction_.squeeze(axis=1)
         if self.oob_prediction_.shape[-1] == 1:
             # drop the n_outputs axis if there is a single output
             self.oob_prediction_ = self.oob_prediction_.squeeze(axis=-1)
-        self.oob_score_ = r2_score(y, self.oob_prediction_)
+        self.oob_score_ = r2_score(
+            y, self.oob_prediction_, sample_weight=sample_weight
+        )
```

Comment:
```suggestion
        if self.oob_score:
            self.oob_score_ = r2_score(
                y, self.oob_prediction_, sample_weight=sample_weight
            )
```

Reply:
It also means that we need to add a test to ensure that `oob_score_` is not set when `oob_score=False` even when the permutation importance is computed.